### ISSUE TITLE
Jumpcut not saving history

### ISSUE THREAD

**Issue by smileymattj** [snark/jumpcut#82]
Hi, I'm running Catalina 10.15.7.  I installed the latest Jumpcut 0.82.  Gave it Accessibility permission.  Drop down menu works.  But when Jumpcut is restarted the history is cleared.  0.82 seem to not be creating the folder: Library/Application\ Support/Jumpcut.  I tried version 0.66 and it created the folder, but is not creating the file JCEngine.save.  

Thanks

**Comment by smileymattj** [comment]
Jumpcut 0.70 & 0.71 are working.

**Comment by snark** [comment]
@smileymattj Thanks for the bug report; I'll check it out.

**Comment by Steeve-S** [comment]
Hi, I'm facing the same issue under Monterey 12.4 and Jumpcut 0.82
Thanks

**Comment by TheOtherLadcd** [comment]
I hate to be "me too", but having recently upgraded from a 2009 Mac Pro to a Studio Max and updating Jumpcut to 0.82, now Jumpcut won't put text copied from 1Password into the history list. It WILL copy/paste the text, just not put it into the history list. Text copied from a Firefox web page does go into the history list.

**Comment by snark** [comment]
@TheOtherLadcd 1Password is a separate issue. Jumpcut follows the NSPasteboard.org community standard of ignoring "transient" values (for instance, those inserted by Keyboard Maestro or TextExpander), and by default Jumpcut doesn't store passwords that come from password managers that correctly identify their fields as "concealed" (e.g. 1Password or MacPass). Try unchecking "ignore confidential clipping types" in the clippings preference and it should do what you want; I've just confirmed this with 1Password 8.7.1 and Jumpcut 0.8.2. (If you upgraded recently, you were probably running an older version of Jumpcut which didn't check for "org.nspasteboard.ConcealedType".)

**Comment by snark** [comment]
@Steeve-S @smileymattj Could one of you give me the terminal output for `ls -ld ~"/Library/Application Support"`? I suspect it's this issue, which seems incredibly sloppy on Apple's part but could well be the case: https://discussions.apple.com/thread/253393011?answerId=256597632022#256597632022

**Comment by TheOtherLadcd** [comment]
Thank you for solving my 1Password copy issue! Only a handful of mysteries left from my huge jump in Mac platforms/OS.

**Comment by Steeve-S** [comment]
Hi @snark here you can find the output for ls -ld ~"/Library/Application Support"
drwx------+ 61 steeve  staff  1952 15 jui 16:48 /Users/steeve/Library/Application Support

and yes you're right there is no Jumcut folder in it, even I've already tried to grant Jumpcut full access to my disk.
ls: /Users/steeve/Library/Application Support/Jumpcut: No such file or directory

I've created the "Jumpcut" folder manually in this path and the problem was solved ! I can see now the "JCEngine.save" file in it. 

Thank you very much ! 

Don't hesitate if you want me to do some others tests for the development of Jumpcut

**Comment by TheOtherLadcd** [comment]
When I followed the suggestion as to how to fix the 1Password issue I was having, everything worked fine. Then I realized that the Jumpcut copy history wasn't being saved between sessions, so I tried the above tip of creating the "Jumpcut" folder manually in Application Support. I too see "JCEngine.save" being created there, so hopefully that solves the history issue also. Thanks for all the tips!  :-)

**Comment by snark** [comment]
@Steeve-S Yeah, that looks right. Thank you! It's possible that I broke something in the Swift transition, but it's also possible that something is busted under Monterey; I'll dig in further, but it's good to know that there's a workaround.

**Comment by pioneer-eleven** [comment]
Hi,
Using Mac OS 10.14.6 here and I think I'm having this issue - only one clip (the last one made with Apple-C appears in the bezel. The Jumpcut menu on the menu bar has all the last several clips but the bezel only has the last one.

Is this the same issue?

**Comment by snark** [comment]
@pioneer-eleven That sounds different to me -- more like it's not receiving keyboard events and redrawing the bezel correctly. Quick questions: what hotkey are you using? And have you tried using the arrows while the bezel is displayed?

**Comment by pioneer-eleven** [comment]
I'm using ctrl-v for paste. Here's the behaviour when I use the arrows...
If I hit ctrl-v and release (normal keystroke with ctrl held) the bezel pops un a disappears immediately the last (and only) clip appears in my text.
If I keep ctrl pressed, the bezel remains on screen but the arrows initiate Spaces desktop switcher tinggy
Sounds like I need to change my shortcut... gimme a mo...

Changed shortcut to shift-ctrl-v - seems to be working. I still only see one clipping in the bezel (I seem to remember I used to get a list), but arrow keys rotate through the last few.

**Comment by snark** [comment]
@pioneer-eleven Oh, I think I understand; when using control-V, if you hold control and repeatedly press "V", is it behaving as expected, cycling through your most recent clips? (In that scenario, while holding control, pressing "V" should move down the stack and pressing "shift-V" should move up.) Holding "shift" is an indicator to move up the stack instead of down, which could potentially lead to some confusion if you're using control-shift-V.

You may want to activate the "sticky bezel" or "wraparound bezel" options, depending on your preferences!
